packages/client/src/hooks/use-wallet.ts

```diff
@@ -132,8 +158,21 @@ export default function useWallet(): {
             providerName: 'metamask',
             provider: (window as any).ethereum,
         };
-        mixpanel.track('wallet', walletObj);
+
         setWallet(walletObj);
+
+        try {
+          const mixpanelData = {
+            distinct_id: account,
+            account,
+            providerName: 'metamask',
+            provider: (window as any).ethereum,
```

Comment:
I would remove these `provider` entries from both this and line 195 - they are not simple string fields, but complex objects, and contain a lot of unneeded data. I think `providerName` tells us all we need to know about the provider from a tracking perspective